agent/packaging/ubuntu_xenial/kontena-agent/lib/systemd/system/kontena-agent.service

```diff
@@ -0,0 +1,26 @@
+[Unit]
+Description=kontena-agent
+After=network-online.target
+After=docker.service
+Description=Kontena Agent
+Documentation=http://www.kontena.io/
+Requires=network-online.target
+Requires=docker.service
+
+[Service]
+Restart=always
+RestartSec=5
+EnvironmentFile=/etc/kontena-agent.env
+ExecStartPre=-/usr/bin/docker stop kontena-agent
+ExecStartPre=-/usr/bin/docker rm kontena-agent
+ExecStartPre=/usr/bin/docker pull kontena/agent:${KONTENA_VERSION}
```

Comment:
This could maybe be prefixed with - to continue trying to run the service, even if the Docker hub is down, or using a local image version...